src/Proto3/Wire/Encode.hs

```diff
@@ -72,34 +79,68 @@ module Proto3.Wire.Encode
     , packedFixed64
     , packedFloats
     , packedDoubles
-      -- * Reexports
-    , Builder
-    , WB.builderLength
-    , WB.rawBuilder
-    , WB.toLazyByteString
     ) where
 
 import           Data.Bits                     ( (.&.), (.|.), shiftL, shiftR, xor )
 import qualified Data.ByteString               as B
+import qualified Data.ByteString.Builder       as BB
 import qualified Data.ByteString.Lazy          as BL
 import           Data.Int                      ( Int32, Int64 )
 import           Data.Monoid                   ( (<>) )
 import qualified Data.Text.Encoding            as Text.Encoding
 import qualified Data.Text.Lazy                as Text.Lazy
 import qualified Data.Text.Lazy.Encoding       as Text.Lazy.Encoding
 import           Data.Word                     ( Word8, Word32, Word64 )
-import           Proto3.Wire.Builder           ( Builder )
 import qualified Proto3.Wire.Builder           as WB
 import           Proto3.Wire.Types
 
--- | $setup
+-- $setup
 --
 -- >>> :set -XOverloadedStrings
 
-base128Varint :: Word64 -> Builder
+-- | A `Message` represents a serialized protobuf message
+--
+-- Use the utilities provided by this module to build `Message`s
+--
+-- You can concatenate two messages using the `Monoid` instance for `Message`
+-- 
+-- Use `toLazyByteString` when you're done assembling the `Message`
+newtype Message = Message { unMessage :: WB.Builder }
+  deriving Monoid
+
+instance Show Message where
+  showsPrec prec builder =
+      showParen (prec > 10)
+        (showString "Proto3.Wire.Encode.unsafeFromLazyByteString " . shows bytes')
+    where
+      bytes' = toLazyByteString builder
+
+-- | Retrieve the length of a message, in bytes
+messageLength :: Message -> Word
+messageLength = WB.builderLength . unMessage
+
+-- | Convert a message to a @"Data.ByteString.Builder".`BB.Builder`@
+rawMessage :: Message -> BB.Builder
+rawMessage = WB.rawBuilder . unMessage
+
+-- | Convert a message to a lazy `BL.ByteString`
+toLazyByteString :: Message -> BL.ByteString
+toLazyByteString = WB.toLazyByteString . unMessage
+
+-- | This is the only utility in this module that lets you build a malformed
+-- protobuf message
```

Comment:
I understand the intent here, but the way it is worded, is this strictly true?  Couldn't one use field numbers inconsistently, for example?